datafusion/optimizer/src/simplify_expressions/regex.rs

```diff
@@ -43,6 +47,23 @@ pub fn simplify_regex_expr(
     let mode = OperatorMode::new(&op);
 
     if let Expr::Literal(ScalarValue::Utf8(Some(pattern))) = right.as_ref() {
+        // Handle the special case for ".*" pattern
+        if pattern == ANY_CHAR_REGEX_PATTERN {
+            let new_expr = if mode.not {
+                // not empty
+                let empty_lit = Box::new(lit(""));
+                Expr::BinaryExpr(BinaryExpr {
+                    left,
+                    op: Operator::Eq,
+                    right: empty_lit,
+                })
+            } else {
+                // always true
```

Comment:
I think this needs to handle nulls too as `null ~ '.*'` is not `true` (it is null)

```sql
> create or replace table foo(x varchar) as values (1), (2), (null);
0 row(s) fetched.
Elapsed 0.004 seconds.

> select x ~ '.*' from foo;
+--------------------+
| foo.x ~ Utf8(".*") |
+--------------------+
| true               |
| true               |
| NULL               |
+--------------------+
3 row(s) fetched.
Elapsed 0.016 seconds.
```

So maybe instead of `lit(true)` it is `x.is_not_null()` 🤔

Reply:
Good catch! I added two cases about null in `simplify_expr.slt`, it should work as expected now.